Bug Report: Failing to update report attached to Semantic Model (DirectLake)

### Library Version

0.1.7

### What is the problem?

Hi guys! First, thanks for amazing work, really loving your library. I've ran into a small issue

I have created Semantic Model with DirectLake connection in my feature workspace. Report is connected to this Semantic Model. After I merge changes to my dev branch, pipeline is triggered. When I use fabric-cicd, I can see that semantic model is published, but when it tries to publish report I get following error

[ERROR] 15:04:56 - Semantic model not found in the repository. Cannot deploy a report with a relative path without deploying the model.

I'm running ItemType = ['SemanticModel','Report']

Please note that Lakehouse I'm refering to in Semantic Model is in totally different workspace.

As said, semantic model is published to workspace and if I try to create report directly in target workspace I'm able to do. But. Report that is in repo is not transfered.

I'm running this on latest release of fabric-cicd.

My Repo looks like this, I have both .SemanticModel and .Report subfolders and I believe that definition.pbir is pointing to the right semantic model.

Is this something that is not supported just yet, should I raise a ticket on your git? I'm just courious, because maybe I'm doing something wrong.

If I try to simply connect to this repo within Fabric, it works without any problems. Down is a screenshot of workspace not connected to DevOps Repo (tried to publish with Fabric ci-cd), and another, just connecting to DevOps Repo.



Thanks in advance!

[fabric_cicd.error.log](https://github.com/user-attachments/files/19016521/fabric_cicd.error.log)

![Image](https://github.com/user-attachments/assets/b3c1efc0-dc93-4d0c-8526-78ef9cb526a5)

![Image](https://github.com/user-attachments/assets/78eaa3f5-22bb-414d-96ce-56d8042e922f)

![Image](https://github.com/user-attachments/assets/64288c36-f428-4e1a-8d35-87ef03a76dde)

![Image](https://github.com/user-attachments/assets/bb4f6e5f-347a-4498-a0c2-37769cb4d45c)

### Is there an ideal solution?

_No response_

### Additional context, screenshots, logs, error output, etc

[Report Repo.zip](https://github.com/user-attachments/files/19016427/Report.Repo.zip)

--- a/docs/how_to/parameterization.md
+++ b/docs/how_to/parameterization.md
@@ -5,12 +5,12 @@ To handle environment-specific values committed to git, use a `parameter.yml` fi
 Example of parameter.yml location based on provided repository directory:
 
 ```python
-    from fabric_cicd import FabricWorkspace
-    workspace = FabricWorkspace(
-        workspace_id="your-workspace-id",
-        repository_directory="C:/dev/workspace",
-        item_type_in_scope=["Notebook"]
-    )
+from fabric_cicd import FabricWorkspace
+workspace = FabricWorkspace(
+    workspace_id="your-workspace-id",
+    repository_directory="C:/dev/workspace",
+    item_type_in_scope=["Notebook"]
+)
 ```
 
 ```

--- a/docs/how_to/parameterization.md
+++ b/docs/how_to/parameterization.md
@@ -22,8 +22,6 @@ C:/dev/workspace
     /parameter.yml
 ```
 
-The `parameter.yml` file should be located directly within the `reposistory_directory` field passed into the `FabricWorkspace` object.
-
 Raise a [feature request](https://github.com/microsoft/fabric-cicd/issues/new?template=2-feature.yml) for additional parameterization capabilities.
 
 ## Inputs

--- a/docs/how_to/parameterization.md
+++ b/docs/how_to/parameterization.md
@@ -1,6 +1,6 @@
 # Parameterization
 
-To handle environment-specific values committed to git, use a `parameter.yml` file. This file supports programmatically changing values based on the `environment` field passed into the `FabricWorkspace` object. If the environment value is not found in the `parameter.yml` file, any dependent replacements will be skipped. This file should sit within the same `repository_directory` folder specified in the FabricWorkspace object.
+To handle environment-specific values committed to git, use a `parameter.yml` file. This file supports programmatically changing values based on the `environment` field passed into the `FabricWorkspace` object. If the environment value is not found in the `parameter.yml` file, any dependent replacements will be skipped. This file should sit in the root of the `repository_directory` folder specified in the FabricWorkspace object.
 
 Example of parameter.yml location based on provided repository directory:
 

--- a/docs/how_to/parameterization.md
+++ b/docs/how_to/parameterization.md
@@ -1,6 +1,28 @@
 # Parameterization
 
-To handle environment-specific values committed to git, use a `parameter.yml` file. This file supports programmatically changing values based on the `environment` field passed into the `FabricWorkspace` object. If the environment value is not found in the `parameter.yml` file, any dependent replacements will be skipped. This file should sit within the same `workspace` folder specified in the FabricWorkspace object.
+To handle environment-specific values committed to git, use a `parameter.yml` file. This file supports programmatically changing values based on the `environment` field passed into the `FabricWorkspace` object. If the environment value is not found in the `parameter.yml` file, any dependent replacements will be skipped. This file should sit within the same `repository_directory` folder specified in the FabricWorkspace object.
+
+Example of parameter.yml location based on provided repository directory:
+
+```python
+    from fabric_cicd import FabricWorkspace
+    workspace = FabricWorkspace(
+        workspace_id="your-workspace-id",
+        repository_directory="C:/dev/workspace",
+        item_type_in_scope=["Notebook"]
+    )
+```
+
+```
+C:/dev/workspace
+    /HelloWorld.Notebook
+        ...
+    /GoodbyeWorld.Notebook
+        ...
+    /parameter.yml
+```
+
+The `parameter.yml` file should be located directly within the `reposistory_directory` field passed into the `FabricWorkspace` object.
 
 Raise a [feature request](https://github.com/microsoft/fabric-cicd/issues/new?template=2-feature.yml) for additional parameterization capabilities.
 

--- a/src/fabric_cicd/_common/_validate_input.py
+++ b/src/fabric_cicd/_common/_validate_input.py
@@ -74,20 +74,27 @@ def validate_item_type_in_scope(input_value: list, upn_auth: bool) -> list:
     return input_value
 
 
-def validate_repository_directory(input_value: str) -> str:
+def validate_repository_directory(input_value: str) -> Path:
     """
-    Validate the repository directory.
+    Validate the repository directory and convert string to Path object
 
     Args:
         input_value: The input value to validate.
     """
     validate_data_type("string", "repository_directory", input_value)
 
-    if not Path(input_value).is_dir():
+    directory = Path(input_value)
+
+    if not directory.is_dir():
         msg = f"The provided repository_directory '{input_value}' does not exist."
         raise InputError(msg, logger)
 
-    return input_value
+    if not directory.is_absolute():
+        absolute_directory = directory.resolve()
+        logger.info(f"Relative directory path '{directory}' resolved as '{absolute_directory}'")
+        directory = absolute_directory
+
+    return directory
 
 
 def validate_base_api_url(input_value: str) -> str:

--- a/src/fabric_cicd/fabric_workspace.py
+++ b/src/fabric_cicd/fabric_workspace.py
@@ -102,7 +102,7 @@ class FabricWorkspace:
 
         # Validate and set class variables
         self.workspace_id = validate_workspace_id(workspace_id)
-        self.repository_directory = validate_repository_directory(repository_directory)
+        self.repository_directory: Path = validate_repository_directory(repository_directory)
         self.item_type_in_scope = validate_item_type_in_scope(item_type_in_scope, upn_auth=self.endpoint.upn_auth)
         self.base_api_url = f"{validate_base_api_url(base_api_url)}/v1/workspaces/{workspace_id}"
         self.environment = validate_environment(environment)
@@ -114,12 +114,12 @@ class FabricWorkspace:
 
     def _refresh_parameter_file(self) -> None:
         """Load parameters if file is present."""
-        parameter_file_path = Path(self.repository_directory, "parameter.yml")
+        parameter_file_path = self.repository_directory / "parameter.yml"
         self.environment_parameter = {}
 
-        if Path(parameter_file_path).is_file():
+        if parameter_file_path.is_file():
             logger.info(f"Found parameter file '{parameter_file_path}'")
-            with Path.open(parameter_file_path, encoding="utf-8") as yaml_file:
+            with parameter_file_path.open(encoding="utf-8") as yaml_file:
                 self.environment_parameter = yaml.safe_load(yaml_file)
 
     def _refresh_repository_items(self) -> None:
@@ -130,7 +130,7 @@ class FabricWorkspace:
             directory = Path(root)
             # valid item directory with .platform file within
             if ".platform" in files:
-                item_metadata_path = Path(directory, ".platform")
+                item_metadata_path = directory / ".platform"
 
                 # Print a warning and skip directory if empty
                 if not os.listdir(directory):
@@ -224,6 +224,7 @@ class FabricWorkspace:
                 if key in raw_file and self.environment in parameter_dict:
                     # replace any found references with specified environment value
                     raw_file = raw_file.replace(key, parameter_dict[self.environment])
+                    logger.debug(f"Replaced '{key}' with '{parameter_dict[self.environment]}'")
 
         return raw_file
 
@@ -318,7 +319,7 @@ class FabricWorkspace:
             path: Full path of the desired item.
         """
         for item_details in self.repository_items[item_type].values():
-            if Path(item_details.path) == Path(path):
+            if item_details.path == Path(path):
                 return item_details.logical_id
         # if not found
         return None

